Localitos/pluto
# chore: commitlint + lint-staged files pre-commit

## Description of the change

Closes #6 

This adds pre-commit checks to ensure that:
 - The commit message conforms to the conventional commit format
 - Staged files don't include lint errors (they will be auto-fixed when possible)

## Testing the change

- Install new deps locally with `npm install`
  - This should set up husky automatically via the `prepare` script. Look for a `./husky` directory to be added to the root
- **Commitlint**
  - Edit and stage any file and attempt to commit with an invalid commit message e.g. `git commit -m "xyz"`
    - This should be blocked and you should see a nice error message explaining why
  - Valid commit messages should work
- **lint-staged** (recommend turning off format on save settings for testing)
  - Stage changes with formatting and/or lint errors.
  - It should run both prettier and eslint and it should fix anything it can automatically. If it runs into a lint error it can't save it will prevent the commit and print the error

## Type of change

- [x] Non-Breaking Change (change to existing functionality)

### Development
N/A

### Code review
N/A
- [ ] Issue from Linear has a link to this pull request
